pkg/lca/ibu_test.go

```diff
@@ -0,0 +1,693 @@
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeWithSeedImagePullSecretRef(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		pullSecretName      string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			pullSecretName:      "pull-secret",
+		},
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			pullSecretName:      "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testIBU := generateImageBasedUpgrade()
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIBU)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		ibuBuilder, err := PullImageBasedUpgrade(testSettings)
+		assert.Nil(t, err)
+
+		// Test the WithExtraManifests function
+		builderResult := ibuBuilder.WithSeedImagePullSecretRef(
+			testCase.pullSecretName)
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeWaitUntilStageComplete(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		stage               string
+	}{
+		{
+			expectedError:       fmt.Errorf("wrong stage selected for imagebasedupgrade"),
+			addToRuntimeObjects: true,
+			stage:               "wrong",
+		},
+		{
+			expectedError:       fmt.Errorf("wrong stage selected for imagebasedupgrade"),
+			addToRuntimeObjects: true,
+			stage:               "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testIBU := generateImageBasedUpgrade()
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIBU)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		ibuBuilder, err := PullImageBasedUpgrade(testSettings)
+		assert.Nil(t, err)
+
+		// Test the WithExtraManifests function
+		builderResult, err := ibuBuilder.WaitUntilStageComplete(
+			testCase.stage)
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeWithStage(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		stage               string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			stage:               "Idle",
+		},
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			stage:               "Wrong",
+		},
```

Comment:
Not sure if you really need this. It doesn't really validate anything different than the first test. Unless you are expecting to change it in the future

Reply:
changed.